File: 13/Solution.java
class Solution {
    public int romanToInt(String s) {
        int num=0;
		for (int i = 0; i < s.length(); i++) {
			char ch = s.charAt(i);
			if(ch=='M') num+=1000;
			else if(ch=='D') num+=500;
			else if(ch=='C') {
				if(i<s.length()-1) {
					if(s.charAt(i+1)=='D') {i++;num+=400;}
					else if(s.charAt(i+1)=='M') {i++;num+=900;}
					else num+=100;
				}else num+=100;
			} 
			else if(ch=='L') num+=50;
			else if(ch=='X') {
				if(i<s.length()-1) {
					if(s.charAt(i+1)=='L') {i++;num+=40;}
					else if(s.charAt(i+1)=='C') {i++;num+=90;}
					else num+=10;
				}else num+=10;
			}
			else if(ch=='V') num+=5;
			else {
				if(i<s.length()-1) {
					if(s.charAt(i+1)=='V') {i++;num+=4;}
					else if(s.charAt(i+1)=='X') {i++;num+=9;}
					else num+=1;
				} else num+=1;
			}
		}
		return num;
    }
}

File: 13/Solution.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        result = 0
        index = 0
        while True:
            flag=True
            if s[index]=='I':
                if index+1<len(s):
                    if s[index+1]=='V':
                        result+=4
                        flag = False
                    elif s[index+1]=='X':
                        result+=9
                        flag=False
                if flag:
                    result+=1
                    index+=1
                else:
                    index+=2
            elif s[index]=='X':
                if index+1<len(s):
                    if s[index+1]=='L':
                        result+=40
                        flag = False
                    elif s[index+1]=='C':
                        result+=90
                        flag=False
                if flag:
                    result+=10
                    index+=1
                else:
                    index+=2
            elif s[index]=='C':
                if index+1<len(s):
                    if s[index+1]=='D':
                        result+=400
                        flag = False
                    elif s[index+1]=='M':
                        result+=900
                        flag=False
                if flag:
                    result+=100
                    index+=1
                else:
                    index+=2
            elif s[index]=='V':
                result+=5
                index+=1
            elif s[index]=='L':
                result+=50
                index+=1
            elif s[index]=='D':
                result+=500
                index+=1
            elif s[index]=='M':
                result+=1000
                index+=1
            if index>=len(s):
                break
        return result


